### PR TITLE
Cleaned up permissions on FilOz-maintained repos

### PR DESCRIPTION
**Pull request by BigLep**
## Summary
Cleaned up permissions on FilOz-maintained repos.

This boiled down to either setting lotus-maintainers or actors-maintainers as the admin for the repos FilOz owns or maintains.

In addition, I generally removed individual repo collaborator permissions to instead rely on team permissions.

## Why do you need this?
Two reasons:
1. I wanted to have an authoritative place to answer / show which repos FilOz maintains.
2. I got annoyed seeing various slack messages about FilOz team members not having permissions to approve/merge on repos they are tasked with maintaining for the ecosystem.

While identifying the FilOz-maintained repos, I also cleaned up permissions.  

## Overview of changes
This is the breakout of the changes I made:

Repos:
* actors-private
   * removed individual collaborators
   * made actors-maintainers the admin
* actors-utils
   * removed most collaborators
   * made actors-maintainers the admin
* builtin-actor-priv
   * removed most collaborators
   * made actors-maintainers the admin
* builtin-actors-bundler
   * removed most collaborators
   * made actors-maintainers the admin
* builtin-actors
   * gave push access to newly created actors-contributors team
   * removed collaborators (relying on teams instead)
* ent
   * removed collaborators (relying on actors-maintainers instead)
* filecoin-actor-utils
   * archived the repo since with now have actors-utils instead
* filecoin-ffi-private
   * removed collaborators
   * made lotus-maintainers and proofs the admin
* filecoin-ffi
   * removed collaborators (relying on teams instead)
* fvm-workbench
   * removed some collaborators
* go-address
   * removed individual collaborators
   * made lotus-maintainers the admin
* go-amt-ipld
   * removed collaborators (relying on teams instead)
* go-bitfield
   * removed individual collaborators
* go-cbor-util
   * made lotus-maintainers the admin
* go-commp-utils
   * removed collaborators
   * made lotus-maintainers and Curio the admin
* go-crypto
   * removed collaborators
   * made lotus-maintainers the admin
* go-data-segment
   * removed some collaborators
* go-ds-versioning
   * removed collaborators
   * made lotus-maintainers the admin
* go-fs-versioning
   * removed most collaborators
   * made lotus-maintainers the admin
* go-fil-commcid-priv
   * removed some collaborators
   * made lotus-maintainers the sole admin
* go-fil-commcid
   * removed most collaborators
   * made lotus-maintainers the sole admin
* go-hamt-ipld
   * removed collaborators (relying on teams instead)
* go-jsonrpc
   * removed collaborators (relying on teams instead)
* go-padreader
   * removed collaborators
   * made lotus-maintainers the sole admin
* go-paramfetch
   * made lotus-maintainers the sole admin
* go-state-types
   * removed collaborators (relying on teams instead)
* go-statemachine
   * removed collaborators
   * made lotus-maintainers and Curio the admin
* go-statestore
   * added lotus-maintainers as the sole admin
* go-storedcounter
   * removed collaborators
   * made lotus-maintainers the admin
* homebrew-lotus
   * removed collaborators (relying on teams instead)
   * made lotus-maintainers the sole admin
* lotus
   * removed individual collaborators but added @sbadrus (others relying on teams instead)
* lotus-archived
   * removed collaborators (relying on teams instead)
   * made lotus-maintainers the sole admin
* lotus-infra
   * removed most collaborators (relying on teams instead)
   * made lotus-maintainers the sole admin
* lotus-priv
   * removed collaborators (relying on teams instead)
   * made lotus-maintainers the sole admin
* pubsub
   * removed collaborators (relying on teams instead)
   * made lotus-maintainers the sole admin
* ref-fvm-priv
   * removed collaborators
   * made lotus-maintainers the admin 
* ref-fvm
   * removed collaborators
   * made lotus-maintainers the admin 
   * added FIL-B for push access
   
   
Teams:
* actors-contributors
   * created this team based on the individual collaborators who had been part of builtin-actors. 
* bloxico-lotus
   * removed the team since didn't have permissions to any repos and so it doesn't show up when searching "lotus"
* lotus-maintainers
   * removed @snadrus from the team (but per above still made sure has maintain permissions on lotus)
* lotus-infra-core
   * removed the team since didn't have permissions to any repos and so it doesn't show up when searching "lotus"
* Strike Team
   * removed since no longer has permissions on any repos as a result of some of the above changes
* strike-team-core
   * removed since doesn't have any permissions.  (It wasn't affected by above changes, but seemed reasonable to do given removing "Strike Team")

## Preconditions before merging this PR
- [x] @BigLep Explain why gave lotus/actors-maintainers "admin" vs. "maintain" permissions: https://github.com/filecoin-project/github-mgmt/pull/50#issuecomment-2266267117
- [x] @BigLep @mention all the people who are affected: https://github.com/filecoin-project/github-mgmt/pull/50#issuecomment-2274133671
- [ ] @BigLep Wait until at least 2024-08-08T21:00:00Z before merging (to give time for any major outside feedback).

## Reviewer's Checklist

- [x] It is clear where the request is coming from (if unsure, ask)
- [x] All the automated checks passed
- [x] The YAML changes reflect the summary of the request
- [x] The Terraform plan posted as a comment reflects the summary of the request

### CODE REVIEW

**Review by anorth — APPROVED** [review]
LGTM, but I'd suggest waiting for at least one more approval.

**Review by rvagg — APPROVED** [review]

**Comment by BigLep** [comment]
[@mentioned below the folks below who are having their permissions affected so they get a notification about this change]

If you are receiving this message, your permissions to repos that [FilOz](https://filoz.org) maintains are being adjusted.  

https://github.com/filecoin-project/github-mgmt/pull/50 provides more context for the change.  It also summarizes the changes that were made at the repo/team level.

An individual-collaborator-focused view of the changes (i.e., what chages are happening for _me_?) is viewable at https://github.com/filecoin-project/github-mgmt/pull/50#issuecomment-2266279597

Feedback or questions welcome.

I'll plan on merging this around 21:00UTC on Thursday, 2024-08-08 if there no major issues are raised by then.  If we messed up anything here, we can revert/amend as needed.  

Thanks,
Steve

<details><summary>Individuals Affected</summary>

@aarshkshah1992
@adlrocha
@alanshaw
@alexytsu
@ameanasad
@androowoo
@andyschwab
@anorth
@arajasek
@aschmahmann
@ashwanth-eth
@aysmed
@biglep
@birdychang
@blocksonachain
@bradholden
@brendalee
@cbrocoum
@cryptonemo
@davidd8
@diegorbaquero
@dkkapur
@drpetervannostrand
@ec2
@elijaharita
@emilymvaughan
@eshon
@figureouter
@fridrik01
@frrist
@galargh
@gammazero
@glacierwalrus
@guanzo
@hammertoe
@hannahhoward
@ianconsolata
@irenegia
@ischasny
@jacarte
@jbenet
@jennijuju
@jsoares
@juliangruber
@kkarrancsu
@kubuxu
@laurenspiegel
@lexluthr
@lucaniz
@maciejwitowski
@macro-ss
@magik6k
@marco-storswift
@masih
@matejpavlovic
@mf416
@mishmosh
@momack2
@nemanjalu92
@nicola
@ninitrava
@nonsense
@ns4plabs
@ognots
@olizilla
@patrickwoodhead
@pk-controller
@pk13-plf
@pl-kevin
@porcuquine
@protocolin
@ranchalp
@raulk
@reiers
@ribasushi
@rjan90
@rk-rishikesh
@rvagg
@scotthconner
@snadrus
@snissn
@stebalien
@stefaan-v
@tchataigner
@terryhung
@tippyflitsuk
@torfinnolsen
@trruckerfling
@tyshko5
@vesahc
@vkalghatgi
@vmx
@whyrusleeping
@willscott
@xbalbinus
@xors-eng
@yiannisbot
@zenground0
</details>

**Comment by Stebalien** [comment]
You hit the notification limit in that comment, so I copied the rest here.

<details><summary>Individuals Affected</summary>
@macro-ss
@magik6k
@marco-storswift
@masih
@matejpavlovic
@mf416
@mishmosh
@momack2
@nemanjalu92
@nicola
@ninitrava
@nonsense
@ns4plabs
@ognots
@olizilla
@patrickwoodhead
@pk-controller
@pk13-plf
@pl-kevin
@porcuquine
@protocolin
@ranchalp
@raulk
@reiers
@ribasushi
@rjan90
@rk-rishikesh
@rvagg
@scotthconner
@snadrus
@snissn
@stebalien
@stefaan-v
@tchataigner
@terryhung
@tippyflitsuk
@torfinnolsen
@trruckerfling
@tyshko5
@vesahc
@vkalghatgi
@vmx
@whyrusleeping
@willscott
@xbalbinus
@xors-eng
@yiannisbot
@zenground0
</details>

**Review by Stebalien — APPROVED** [review]

**Comment by BigLep** [comment]
> You hit the notification limit in that comment, so I copied the rest here.

Wow - great catch - good to know - thanks!

File: github/filecoin-project.yml
Comment (masih) [inline github/filecoin-project.yml:2522]: At least one active maintainer of `go-f3` either @Stebalien, @kubuxu or @masih should be admin.

File: github/filecoin-project.yml
Comment (masih) [inline github/filecoin-project.yml:2524]: F3 engineering team should have maintain access: @Stebalien, @kubuxu, @masih.

File: github/filecoin-project.yml
Comment (masih) [inline github/filecoin-project.yml:2447]: F3 engineering team should have push access: @Stebalien, @Kubuxu, @masih, even maybe the entire FilOz if there is an associated GitHub team.

**Review by masih — CHANGES_REQUESTED** [review]

File: github/filecoin-project.yml
Comment (BigLep) [reply github/filecoin-project.yml:2447]: Per below (I know GitHub is hiding it), lotus-maintainers has admin so I think we're good here.

File: github/filecoin-project.yml
Comment (BigLep) [reply github/filecoin-project.yml:2522]: I know the GitHub diff doesn't make this clear, but what happened here is that lotus-maintainers has admin and I removed individuals (besides anorth since he isn't in lotus-maintainers).  As a result I think we're good.

File: github/filecoin-project.yml
Comment (BigLep) [reply github/filecoin-project.yml:2524]: Per comments above/below, we're covered by lotus-maintainers having admin.

File: github/filecoin-project.yml
Comment (masih) [reply github/filecoin-project.yml:2447]: Ah; thanks

**Comment by Kubuxu** [comment]
I see that we are cleaning up permissions on many `-priv` repos. Please ensure that they still have proper access for people who require them. Access to them is usually on a need-basis.

**Comment by Stebalien** [comment]
@Kubuxu you can see the exact changes in https://github.com/filecoin-project/github-mgmt/pull/50#issuecomment-2266279597.

But you bring up a good point, *-priv repos should probably have no group access and instead declare individual users.

**Comment by BigLep** [comment]
> *-priv repos should probably have no group access and instead declare individual users

I'm happy to do that in a followup if you'd like so I can get things unblocked for now.  Here is what things looked like before.   Feedback welcome on how to adjust, or I can just return to no teams and to use these original individuals.

builtin-actors-priv:
```
-    collaborators:
-      admin:
-        - momack2
-        - ZenGround0
-      maintain:
-        - LaurenSpiegel
-      pull:
-        - snissn
-      push:
-        - arajasek
-        - Kubuxu
```

actors-private:
```
-    collaborators:
-      admin:
-        - momack2
-        - ZenGround0
-      push:
-        - arajasek
```

filecoin-ffi-private:
```
-    collaborators:
-      admin:
-        - cryptonemo
```

go-fil-commcid-priv:
```
-      admin:
-        - arajasek
+      maintain:
         - aschmahmann
-        - davidd8
-        - Kubuxu
         - ribasushi
```

lotus-priv:
```
-    collaborators:
-      admin:
-        - arajasek
-        - jennijuju
-        - Stebalien
-      maintain:
-        - Kubuxu
-      push:
-        - pl-kevin
```

ref-fvm-priv:
```
-    collaborators:
-      admin:
-        - Stebalien
```

**Comment by Stebalien** [comment]
Yep, that's fair. There's nothing _super_ private in them at the moment, IIRC.

**Pull request merged**
